polaris-react/src/components/IndexTable/IndexTable.tsx

```diff
@@ -250,6 +251,13 @@ function IndexTableBase({
     handleCanScrollRight();
   }, [handleCanScrollRight]);
 
+  const handleIsSmallScreen = useCallback(() => {
+    const newSmallScreen = isSmallScreen();
+    if (smallScreen !== newSmallScreen) {
+      setSmallScreen(newSmallScreen);
+    }
```

Comment:
I think we can remove the conditional here because states only get updated if the value passed to the setter method is not the same as the current value.

```suggestion
    setSmallScreen(isSmallScreen());
```